### PR TITLE
CIR-643 added check for 3 deemed parents to continue on

### PR DESCRIPTION
**Pull request by funkmaster20th**

### CODE REVIEW

**Comment by funkmaster20th** [comment]
ready for review

File: test/controllers/groupStructure/DeemedParentReviewAnswersListControllerSpec.scala [hunk 1]
```diff
@@ -110,7 +111,29 @@ DeemedParentReviewAnswersListControllerSpec extends SpecBase with FeatureSwitchi
...
           status(result) mustBe SEE_OTHER
-          redirectLocation(result) mustBe Some("/foo")//todo check this
+          redirectLocation(result) mustBe Some(FakeGroupStructureNavigator.nextPage(
+            page = GroupRatioElectionPage,
+            mode = NormalMode,
+            userAnswers = emptyUserAnswers).url)
+        }
+      }
+
+      "when there is 3 deemed parents" should {
```
Comment (LadonJackson) [inline test/controllers/groupStructure/DeemedParentReviewAnswersListControllerSpec.scala]: This says it's testing that you have 3 deemed parents, but the test itself just has empty user answers and then enters "no" on the form which just so happens to have the same result. Instead I think you need to add 3 deemed parents and then submit the fake request without any answer

File: app/controllers/groupStructure/DeemedParentReviewAnswersListController.scala [hunk 2]
```diff
@@ -64,14 +64,17 @@ class DeemedParentReviewAnswersListController @Inject()(override val messagesApi
...
     formProvider().bindFromRequest().fold(
       formWithErrors =>
         Future.successful(BadRequest(renderView(formWithErrors)))
       ,
       {
         case true => Future.successful(Redirect(navigator.addParent(deemedParents.length)))
         case false => saveAndRedirect(DeemedParentPage, NormalMode)
+      })
+      }
+      else{saveAndRedirect(DeemedParentPage, NormalMode)
```
Comment (LadonJackson) [inline app/controllers/groupStructure/DeemedParentReviewAnswersListController.scala]: This formatting hurts my eyes

File: test/controllers/groupStructure/DeemedParentReviewAnswersListControllerSpec.scala (hunk 1 above)
Comment (funkmaster20th) [reply test/controllers/groupStructure/DeemedParentReviewAnswersListControllerSpec.scala]: updated the test.

File: app/controllers/groupStructure/DeemedParentReviewAnswersListController.scala (hunk 2 above)
Comment (funkmaster20th) [reply app/controllers/groupStructure/DeemedParentReviewAnswersListController.scala]: moved around some of the things

**Pull request merged**